torch/_inductor/ir.py

```diff
@@ -5094,6 +5094,23 @@ def get_layout(self) -> Layout:
             return super().get_layout()
 
 
+class CuteDSLTemplateBuffer(TemplateBuffer):
```

Comment:
I noticed the `TritonTemplateBuffer` has some mutated inputs handling in the constructor, is that needed here?

Reply:
Yeah good call, let me roll that in from here: https://github.com/drisspg/pytorch/pull/2/